internal/extension/proto/extension_service.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoygateway.extension;
+
+import "envoy/config/cluster/v3/cluster.proto";
+import "envoy/config/listener/v3/listener.proto";
+import "envoy/config/route/v3/route.proto";
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "internal/extension/proto/extension_context.proto";
+
+option go_package = "internal/extension/proto";
+
+service EnvoyGatewayExtension {
+    rpc PostHTTPListenerTranslation (PostHTTPListenerTranslationRequest) returns (PostHTTPListenerTranslationResponse) {};
```

Comment:
The idea here is that the implementation of `XDSHookClient.PostHTTPListenerTranslation` would iterate through the resource table and send a unary request for each listener that contains extensionRefs. That could be done as separate goroutines if we want to parallelize it.